hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneAddress.java

```diff
@@ -96,14 +108,38 @@ public void checkUrlTypes() throws OzoneClientException, IOException {
     Assert.assertEquals("key1/key3/key", address.getKeyName());
     Assert.assertFalse("this should not be a prefix",
         address.isPrefix());
+  }
 
+  @Test
+  public void checkPrefixUrlType() throws OzoneClientException {
     address = new OzoneAddress(prefix + "vol1/bucket/prefix");
     address.ensurePrefixAddress();
     Assert.assertEquals("vol1", address.getVolumeName());
     Assert.assertEquals("bucket", address.getBucketName());
     Assert.assertEquals("prefix", address.getKeyName());
     Assert.assertTrue("this should be a prefix",
         address.isPrefix());
+  }
+
+  @Test
+  public void checkSnapshotUrlType() throws OzoneClientException {
+    address = new OzoneAddress(prefix + "vol1/bucket/.snapshot/snap1");
+    address.ensureSnapshotAddress();
+    Assert.assertEquals("vol1", address.getVolumeName());
+    Assert.assertEquals("bucket", address.getBucketName());
+    Assert.assertEquals(".snapshot/snap1", address.getSnapshotName());
+    Assert.assertEquals(".snapshot/snap1", address.getKeyName());
+
+
+    String message = "Delimiters (/) not allowed following " +
```

Comment:
NIT:  if you are going to check for a precise error message, I would define it in a string constant in the class being tested, and then use that string constant in the test.  DRY!

Reply:
@GeorgeJahad It's snapshot specific and it won't be reused in another method, so I simplified it and kept only the snapshot part.